k3k-kubelet/controller/syncer/persistentvolumeclaims.go

```diff
@@ -136,3 +152,81 @@ func (r *PVCReconciler) pvc(obj *v1.PersistentVolumeClaim) *v1.PersistentVolumeC
 
 	return hostPVC
 }
+
+func (r *PVCReconciler) pseudoPV(ctx context.Context, pvc v1.PersistentVolumeClaim, log logr.Logger) error {
```

Comment:
I think we should rename this method to something more explicit, like `createVirtualPersistentVolume`, and add a nice and deep comment on why this is needed.